### DISCUSSION TITLE
Using enums defined in prisma.schema to create typegraphql types?

### DISCUSSION THREAD

**Discussion by kleydon** [MichalLytek/typegraphql-prisma#360] [Q&A]
Is there a clean way to use enums defined in `prisma.schema`, within types created using `typegraphql`? (I've scoured the `typegraphql` and the `typegraphql-prisma` docs, but am not yet seeing the "right" way to do this...)

I'm trying to create the following type (+ input type) :

```ts
import { SoundTypeId } from '@prisma/client'

@ObjectType()
@InputType('SoundTypeInput')
export class SoundType {

  @Field(() => SoundTypeId)  // SoundTypeId from @prisma/client
  soundTypeId!: SoundTypeId

  ...
}
```

...where `SoundTypeId` is defined within `prisma.schema`:

```ts
enum SoundTypeId {
  BackgroundNoise
  ForegroundNoise
  Other
}
```

When I run `prisma generate` to create my `prisma client` and invoke the typegraphql-prisma plugin, I get the error:

```
CannotDetermineGraphQLTypeError: Cannot determine GraphQL output type 
for 'soundTypeId' of 'SoundType' class. Is the value, that is used as its TS type 
or explicit type, decorated with a proper decorator or is it a proper output value?
```

When I then try to register `SoundTypeId` with type-graphql using `registerEnumType()`,

```ts
registerEnumType(SoundTypeId, {
  name: 'SoundTypeId'
})
```

...I run into the following error when the schema is constructed in code:

```
Error: Schema must contain uniquely named types but contains multiple types named "SoundTypeId".
```

If I then _change_ the registered name, i.e.

```ts
registerEnumType(SoundTypeId, {
  name: 'TqlSoundTypeId'  // Changed from SoundTypeId
})
```

All errors disappear - but the result yields effectively duplicate enums in `schema.graphql`:

```graphql

enum SoundTypeId {  // The original, generated based on prisma.schema
  BackgroundNoise
  ForegroundNoise
  Other
}

...

enum TqlSoundTypeId {  // The duplicate, generated based on registerEnumType()
  BackgroundNoise
  ForegroundNoise
  Other
}
```

Is there a better way to use enums defined in `prisma.schema` within typegraphql types, which does not involve this kind of duplication?

Thanks,

-K

**Comment by MichalLytek** [comment]
import the generated enum

**Reply by kleydon** [reply]
@MichalLytek can you please elaborate?

At the top of this thread, I've tried 
```ts
import { SoundTypeId } from '@prisma/client'
```
which isn't working, and when I use `registerEnumType()` to make type-graphql aware of `SoundTypeId`, I get 

```
Error: Schema must contain uniquely named types but contains multiple types named "SoundTypeId".
```
Any recommendation?

Thanks for creating this great resource.

**Reply by MichalLytek** [reply]
import the generated enum, from typegraphql-prisma output folder.

**Reply by nathanLR** [reply]
Hello, sorry to bring back this issue but I'm facing a similar problem. I've followed the steps listed in the doc to implement enums but I keep getting the CannotDetermineGraphQLTypeError. I am NOT using prisma but typeorm.
If you need to see the code i'm having an issue with I've posted a question without success [on stackoverflow](https://stackoverflow.com/questions/79031285/cannotdeterminegraphqltypeerror-when-using-enum-and-typegraphql)
